src/HomeWork_7/HomeWork_7.ts

```diff
@@ -0,0 +1,129 @@
+// У вас є дві сутності - список фільмів і список категорій фільмів.
+
+// Кожен фільм містить поля: назва, рік випуску, рейтинг, список нагород.
+
+// Категорія містить поля: назва і фільми.
+
+// У кожного списку є пошук за ім'ям (це, по суті, фільтрація),
+//  у списку фільмів є додаткова фільтрація за роком випуску, рейтингом і нагородами.
+
+// У нас визначено три типи фільтрів:
+
+// Фільтр відповідності має поле filter
+// Фільтр діапазону має поле filter і filterTo
+// Фільтр пошуку за значеннями має поле values
+// Кожен список містить стан його фільтрів, який може бути змінений
+// тільки методом applySearchValue або applyFiltersValue(за наявності додаткових фільтрів)
+
+// Вам необхідно подумати про поділ вашого коду на різні сутності, інтерфеси і
+// типи, щоб зробити ваше рішення типобезпечним.Реалізація всіх методів не є необхідною - це за бажанням.
```

Comment:
And after creating all the interfaces, try to implement them in the classes that need them.